sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -1420,4 +1420,16 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
       }.getMessage should include("No such struct field b in a, B")
     }
   }
+
+  test("SPARK-32521: withField with statically evaluable arguments should succeed") {
```

Comment:
The end users can see the examples. To ensure we do not have a regression, could you add all the examples to the test suite?

Reply:
Done.